mlflow/pyspark/optuna/study.py

```diff
@@ -0,0 +1,97 @@
+import datetime
+import sys
+from collections.abc import Callable, Iterable
+from typing import Optional
+
+try:
+    import optuna
+    from optuna import storages
+    from optuna.study import Study
+    from optuna.study._optimize import _optimize_sequential
+    from optuna.trial import FrozenTrial
+except ImportError:
+    sys.exit()
+
+from pyspark.sql import SparkSession
+
+from mlflow.optuna.storage import MlflowStorage
+
+
+class MLFlowSparkStudy:
+    """A wrapper of :class:`~optuna.study.Study` to incorporate Optuna with spark
+    via MLFlow experiment.
+    """
+
+    def __init__(self, study_name: str, storage: MlflowStorage):
+        self._study_name = study_name
+        self._storage = storages.get_storage(storage)
+
+        self.spark = SparkSession.active()
+
+        if not isinstance(self._storage, MlflowStorage):
+            raise ValueError(f"MLFlowSparkStudy only works with `MlflowStorage`. But get {type(self._storage)}.")
+
+    def optimize(
```

Comment:
Can we add cancellation command handler ?
i.e.,

in notebook , when click command cancellation command,
it should be able to interrupt the `optimize` execution, and cancel all backend spark jobs.

You can use `spark.addTag` and `spark.interruptTag` APIs to do the job cancellation.